ui/src/app/customizations/tech-extensions/import/tech-extension-import.component.ts

```diff
@@ -1,15 +1,111 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, ElementRef, ViewChild } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import {
+  FileUploader,
+  FileItem,
+  FileLikeObject,
+  FilterFunction,
+  FileUploaderOptions,
+  ParsedResponseHeaders
+} from 'ng2-file-upload-base/src';
+import { NotificationType } from 'patternfly-ng';
+import { Extension } from '../../../model';
 import { ExtensionStore } from '../../../store/extension/extension.store';
+import { NotificationService } from 'app/common/ui-patternfly/notification-service';
+
+interface FileError {
+  level: string;
+  message: string;
+}
 
 @Component({
   selector: 'syndesis-tech-extentions-import',
   templateUrl: 'tech-extension-import.component.html'
 })
+export class TechExtensionImportComponent {
+
+  uploader: FileUploader;
+  response: Extension = undefined;
+  error: FileError = undefined;
+  importing = false;
+
+  @ViewChild('fileSelect') fileSelect: ElementRef;
 
-export class TechExtensionImportComponent implements OnInit {
-  constructor(private store: ExtensionStore) { }
+  constructor(private store: ExtensionStore,
+              private notificationService: NotificationService,
+              private router: Router,
+              private route: ActivatedRoute) {
+    this.uploader = new FileUploader({
+      url: this.store.getUploadUrl(),
+      disableMultipart: false,
+      autoUpload: true,
+      filters: [
+        {
+          name: 'filename filter',
+          fn: (item: FileLikeObject, options: FileUploaderOptions) => {
+            if (!item.name.endsWith('.jar')) {
+              return false;
+            }
+            return true;
+          }
+        }
+      ]
+    });
+    this.uploader.onWhenAddingFileFailed = (item: FileLikeObject, filter: any, options: any): any => {
+      this.error = this.getGenericError();
+      this.fileSelect.nativeElement['value'] = '';
+      this.uploader.clearQueue();
+    };
+    this.uploader.onCompleteItem = (item: FileItem, response: string, status: number, headers: ParsedResponseHeaders) => {
+      this.error = undefined;
+      this.response = undefined;
+      let resp: any = {};
+      try {
+        resp = JSON.parse(response);
+      } catch (err) {
+        this.error = this.getGenericError();
+        return;
+      }
+      if (status === 200) {
+        this.response = <Extension> resp;
+        return;
+      }
+      this.error = {
+        level: 'alert alert-danger',
+        message: resp.userMsg || 'An unknown error has occurred'
+      };
+    };
```

Comment:
I'd say this is too much logic for a constructor. I know that constructors are meant to initialize stuff but perhaps this is too much. On the other hand most of this logic seems that might be of interest for other components that require upload functionalities.

Perhaps it would be worth to promote this to a parent class this component might extend from or even an injectable service that receives the uploader, error, response and fileSelect in its initialize payload.

Reply:
All this code is basically for initializing the file upload widget, first creating an instance and then overriding some handlers.  I could move it to `ngOnInit` I guess.